gtsam/discrete/DecisionTree.h

```diff
@@ -79,7 +79,13 @@ namespace gtsam {
       const void* id() const { return this; }
 
       // everything else is virtual, no documentation here as internal
-      virtual void print(const std::string& s = "") const = 0;
+      virtual void print(
+          const std::string& s = "",
+          const std::function<std::string(L)> formatter = [](const L& x) {
```

Comment:
Hmm. Look at how Key does it? This clutters the method declaration…

Reply:
Added a `defaultFormatter` method in the protected namespace of the class.